### ISSUE TITLE
CTRL+C when using ZREP_RESUME=yes shouldn't clear properties

### ISSUE THREAD

**Issue by darkpixel** [bolthole/zrep#139]
```
root@uspdxopnas01:~# SSH="ssh -p 224" ZREP_SKIP_EXPIRE=1 ZREP_RESUME=yes ZREP_R=-R ZREP_INC_FLAG=-i ZREP_SEND_FLAGS="-v" /usr/local/bin/zrep -t zrep-remote init tank/3dimages --redacted-- tank/backups/uspdxop/3dimages
Setting zrep properties on tank/3dimages
Creating snapshot tank/3dimages@zrep-remote_000000
Sending initial replication stream to --redacted--:tank/backups/uspdxop/3dimages
full send of tank/3dimages@zfs-auto-snap_02-2019-07-01-0700 estimated size is 40.6K
send from @zfs-auto-snap_02-2019-07-01-0700 to tank/3dimages@zfs-auto-snap_02-2019-08-01-0700 estimated size is 624B
--snip--
send from @zfs-auto-snap_05-2019-11-23-0000 to tank/3dimages@zrep-remote_000000 estimated size is 624B
total estimated size is 290G
TIME        SENT   SNAPSHOT
16:13:25   3.06M   tank/3dimages@zfs-auto-snap_03-2019-10-29-0700
16:13:26   3.06M   tank/3dimages@zfs-auto-snap_03-2019-10-29-0700
16:13:27   3.06M   tank/3dimages@zfs-auto-snap_03-2019-10-29-0700
16:13:28   3.06M   tank/3dimages@zfs-auto-snap_03-2019-10-29-0700
16:13:29   3.06M   tank/3dimages@zfs-auto-snap_03-2019-10-29-0700
16:13:30   3.06M   tank/3dimages@zfs-auto-snap_03-2019-10-29-0700
16:13:31   3.06M   tank/3dimages@zfs-auto-snap_03-2019-10-29-0700
16:13:32   3.06M   tank/3dimages@zfs-auto-snap_03-2019-10-29-0700
16:13:33   6.57M   tank/3dimages@zfs-auto-snap_03-2019-10-29-0700
16:13:34   8.73M   tank/3dimages@zfs-auto-snap_03-2019-10-29-0700
16:13:35   10.7M   tank/3dimages@zfs-auto-snap_03-2019-10-29-0700
16:13:36   13.0M   tank/3dimages@zfs-auto-snap_03-2019-10-29-0700
16:13:37   15.0M   tank/3dimages@zfs-auto-snap_03-2019-10-29-0700
16:13:38   16.9M   tank/3dimages@zfs-auto-snap_03-2019-10-29-0700
^C
Destroying any zrep-related snapshots from tank/3dimages
Removing zrep-related properties from tank/3dimages
Error: Error transferring tank/3dimages@zrep-remote_000000 to uswuxsdrtr01.welovesmiles.com:tank/backups/uspdxop/3dimages. Resetting
root@uspdxopnas01:~# 
```

Killing the PID of ```zfs resume``` on the remote does the same.

I'm thinking zrep shouldn't reset when this happens.

**Comment by ppbrown** [comment]
Hmmm..
"shouldnt reset"?

I think you're implying a bit more then merely "shouldnt reset".

if you just rerun the same command, I believe the existing code should pick
up where it left off. So in that sense there is no harm done, right?

on the one hand, you might say, "wel, but it would be nice if zrep wold
remember the settings and automatically pick them up again".

Trouble is, it doesnt preserve ALL the settings in the ZFS properties. only
the lowest level ones.
All those extra environment vars you show off in the ticket.... not to
mention ones that OTHER people tend to use, such as
On an automated restart, zrep woud have no idea of those just based on the
current properties saved. ie: BBCP

So without a large-ish rewrite, seems like best safest thing is to preserve
the status quo, no?
Which forces people to have a consistent framework to launch zrep

**Comment by darkpixel** [comment]
> I believe the existing code should pick up where it left off.

It did not, but I'm going to re-check my results to make sure I have a clean test with documentation.

**Comment by darkpixel** [comment]
Bah!  I tracked it down to an out-of-date copy of zrep.
Properties are being kept in the current version.  Sorry for the noise.

**Comment by ppbrown** [comment]
well... erm..
I'm glad its working for you...
but what you described, does not match how I was envisioning it working :}
you said "properties are being kept".
but that doesnt make sense to me.
the latest code calls clearquit() on send fail.
which should clear the zrep variables.
reason being:
if send fails, maybe the values are wrong

Although looks like clearquit goes a bit too far for that case and clears
snapshots as well.
you didnt make a good test. you should kill the send, not control-c zrep.

I'll patch clearquit to not delete snapshot if resume set.

**Comment by darkpixel** [comment]
Hmm....that's entirely possible.  I tried a few ways--killing the PID of zfs receive, killing the PID of zfs send, and I've tried CTRL+C and I'm sure in a few cases I've hammered CTRL+C a few times out of habit, possibly killing zrep before it could do it's thing.

I agree--if ```ZREP_RESUME=yes``` snapshots probably should get deleted and the user should have to manually ```zrep clear tank/whatever``` and reconfigure if they messed up.  If they aren't using ```ZREP_RESUME=yes``` there's no point in keeping the snapshots on init because they never went anywhere.
